src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp

```diff
@@ -47,6 +52,24 @@ void mark_runtime_skippable_nodes::run(program& p) {
             // if node is already optimized at compilation time, do not handle at runtime
             if (node.can_be_optimized())
                 return;
+
+            std::function<bool(const program_node& node)> all_users_are_shape_of = [&](const program_node& node) {
+                if (node.is_input() || node.is_output())
+                    return false;
+                for (auto& u : node.get_users()) {
+                    if (!u->is_type<shape_of>())
+                        return false;
+                }
+                return true;
+            };
+
+            // TODO: if this pattern is observed frequently, to add this as a separate pass.
+            if (!node.is_output() && all_users_are_shape_of(node)) {
```

Comment:
I believe this optimization may be applied to almost all primitive types

Reply:
I think so, except for some exceptional cases like non_zero / nms / unique, etcs...

Reply:
I agree, this could be generalized since pretty much any primitive which has shape_of as its only user can be optimized out.

Reply:
I updated the commit, could you review again?